### ISSUE TITLE
[Bug] Ghost Transactions (txn's not propagating to the network)

### ISSUE THREAD

**Issue by asharma13524** [AleoHQ/snarkOS#2864]
## 🐛 Bug Report

When attempting to deploy a program or execute transitions, they do not appear on-chain despite receiving a "successfully broadcast" message from the snarkos cli.

## Steps to Reproduce

1. Attempt to deploy the following aleo instructions below with `snarkos developer deploy multiparty_pvp_utils_v008.aleo --priority-fee 10000000 --private-key $PK --query https://api.explorer.aleo.org/v1 --broadcast https://api.explorer.aleo.org/v1/testnet3/transaction/broadcast`.  Link to code: https://github.com/puzzlehq/serengeti/blob/main/multiparty_pvp_utils_vXXX/build/main.aleo 

```
program multiparty_pvp_utils_v008.aleo;
struct Mssage:
    field_1 as field;
    field_2 as field;
    field_3 as field;
    field_4 as field;
    field_5 as field;

record Answer:
    owner as address.private;
    challenger_address as address.private;
    opponent_address as address.private;
    game_address as address.private;
    amount as u64.private;
    nonce as field.private;
    answer as field.private;
    message_1 as field.private;
    message_2 as field.private;
    message_3 as field.private;
    message_4 as field.private;
    message_5 as field.private;

record AuditMintAnswer:
    owner as address.private;
    answer_owner as address.private;
    challenger_address as address.private;
    opponent_address as address.private;
    game_address as address.private;
    amount as u64.private;
    nonce as field.private;
    answer as field.private;
    message_1 as field.private;
    message_2 as field.private;
    message_3 as field.private;
    message_4 as field.private;
    message_5 as field.private;

record RevealAnswer:
    owner as address.private;
    challenger_address as address.private;
    opponent_address as address.private;
    game_address as address.private;
    amount as u64.private;
    nonce as field.private;
    answer as field.private;
    sig as signature.private;

record AuditRevealAnswer:
    owner as address.private;
    challenger_address as address.private;
    opponent_address as address.private;
    game_address as address.private;
    amount as u64.private;
    nonce as field.private;
    answer as field.private;
    sig as signature.private;

record Key:
    owner as address.private;
    seed as field.private;
    amount as u64.private;
    challenger as address.private;
    opponent as address.private;
    game_multisig as address.private;

record AuditKey:
    owner as address.private;
    seed as field.private;
    amount as u64.private;
    challenger as address.private;
    opponent as address.private;
    game_multisig as address.private;


function mint_answer:
    input r0 as address.private;
    input r1 as address.private;
    input r2 as address.private;
    input r3 as u64.private;
    input r4 as field.private;
    input r5 as field.private;
    input r6 as field.private;
    input r7 as field.private;
    input r8 as field.private;
    input r9 as field.private;
    input r10 as field.private;
    input r11 as signature.private;
    cast r6 r7 r8 r9 r10 into r12 as Mssage;
    sign.verify r11 r0 r12 into r13;
    assert.eq true r13;
    cast r0 r0 r1 r2 r3 r4 r5 r6 r7 r8 r9 r10 into r14 as Answer.record;
    cast aleo16hf8hfpwasnn9cf7k2c0dllc56nn7qt547qxgvgwu6pznw4trvqsx68kls r0 r0 r1 r2 r3 r4 r5 r6 r7 r8 r9 r10 into r15 as AuditMintAnswer.record;
    output r14 as Answer.record;
    output r15 as AuditMintAnswer.record;


function mint_multisig_key:
    input r0 as field.private;
    input r1 as u64.private;
    input r2 as address.private;
    input r3 as address.private;
    input r4 as address.private;
    cast r3 r0 r1 r2 r3 r4 into r5 as Key.record;
    cast aleo16hf8hfpwasnn9cf7k2c0dllc56nn7qt547qxgvgwu6pznw4trvqsx68kls r0 r1 r2 r3 r4 into r6 as AuditKey.record;
    output r5 as Key.record;
    output r6 as AuditKey.record;


function reveal_answer:
    input r0 as Answer.record;
    input r1 as signature.private;
    assert.eq r0.owner r0.challenger_address;
    cast r0.message_1 r0.message_2 r0.message_3 r0.message_4 r0.message_5 into r2 as Mssage;
    sign.verify r1 r0.challenger_address r2 into r3;
    assert.eq true r3;
    cast r0.game_address r0.challenger_address r0.opponent_address r0.game_address r0.amount r0.nonce r0.answer r1 into r4 as RevealAnswer.record;
    cast aleo16hf8hfpwasnn9cf7k2c0dllc56nn7qt547qxgvgwu6pznw4trvqsx68kls r0.challenger_address r0.opponent_address r0.game_address r0.amount r0.nonce r0.answer r1 into r5 as AuditRevealAnswer.record;
    output r4 as RevealAnswer.record;
    output r5 as AuditRevealAnswer.record;
```
2. See the following logs: 
```
📦 Creating deployment transaction for 'multiparty_pvp_utils_v008.aleo'...

✅ Created deployment transaction for 'multiparty_pvp_utils_v008.aleo'
✅ Successfully broadcast deployment at1a6jpl4ys9r5n5jvmpywnpy8e35rsgdyukllfztqxk0kdczh6f59q54ttjt ('multiparty_pvp_utils_v008.aleo') to https://api.explorer.aleo.org/v1/testnet3/transaction/broadcast.
at1a6jpl4ys9r5n5jvmpywnpy8e35rsgdyukllfztqxk0kdczh6f59q54ttjt
```
3. Upon searching for the transaction on hamp, aleo explorer, or aleo123, it does not appear.

## Expected Behavior

The transaction should show up on chain when receiving a successful response from the snarkos cli.

## Your Environment

-  v2.2.4
-  v1.73.0

**Comment by d0cd** [comment]
I was able to deploy this program using the following command.
```
snarkos developer deploy "multiparty_pvp_utils_v008.aleo" --path "./" --private-key <PK> --query "https://api.explorer.aleo.org/v1" --broadcast "https://api.explorer.aleo.org/v1/testnet3/transaction/broadcast" --priority-fee 0
```
See [here](https://explorer.aleo.org/program/multiparty_pvp_utils_v008.aleo)

To troubleshoot "ghost transactions" please:
- check that you have sufficient fees to pay for the deployment. 
- if you are sure that you have sufficient fees, you can check:
> - the `/testnet3/memoryPool/transactions` endpoint for transaction in the mempool
> - the `/testnet3/transaction/confirmed/:id` endpoint to see whether it was accepted or rejected.

**Comment by asharma13524** [comment]
> I was able to deploy this program using the following command.
> 
> ```
> snarkos developer deploy "multiparty_pvp_utils_v008.aleo" --path "./" --private-key <PK> --query "https://api.explorer.aleo.org/v1" --broadcast "https://api.explorer.aleo.org/v1/testnet3/transaction/broadcast" --priority-fee 0
> ```
> 
> See [here](https://explorer.aleo.org/program/multiparty_pvp_utils_v008.aleo)
> 
> To troubleshoot "ghost transactions" please:
> 
> * check that you have sufficient fees to pay for the deployment.
> * if you are sure that you have sufficient fees, you can check:
> 
> > * the `/testnet3/memoryPool/transactions` endpoint for transaction in the mempool
> > * the `/testnet3/transaction/confirmed/:id` endpoint to see whether it was accepted or rejected.

thanks for the prompt reply! much appreciated.

**Comment by asharma13524** [comment]
@d0cd Thanks for your help here. I was able to deploy another one of our imports. However, It appears I'm running into this issue with the following program and am getting 404s on the memoryPool endpoints. Additionally just getting the generic "Missing transaction ..." via the `/testnet3/transaction/confirmed/:id` endpoint despite snarkos returning a successful response. Please let me know if there is a better way to debug.

https://api.explorer.aleo.org/v1/testnet3/memoryPool/transactions
https://api.explorer.aleo.org/v1/testnet3/memoryPool/transmissions
https://api.explorer.aleo.org/v1/testnet3/memoryPool/solutions

1. Attempt to deploy the following aleo instructions below with 

```
snarkos developer deploy "wheres_alex_v008.aleo" --path "./" --private-key $PK --query "https://api.explorer.aleo.org/v1" --broadcast "https://api.explorer.aleo.org/v1/testnet3/transaction/broadcast" --priority-fee 0
``` 

Link to code: https://github.com/puzzlehq/serengeti/blob/main/wheres_alex_vXXX/build/main.aleo

**Comment by mwyatt896** [comment]
@d0cd following up here as this may be a new bug for ghost transactions

Turns out we had a transition in our puzzle_pieces program that had 3 input records but only had 2 outputs (just 2 records). 

We were able to deploy the puzzle_pieces program with this function to the network and it wasn't causing errors until using the program as an import in our wheres_alex program.

The details can be seen in our puzzle_pieces_v009.aleo on our joint_stake_transfer_to_winner function [here](https://explorer.hamp.app/program?id=puzzle_pieces_v009.aleo).

`snarkOS developer deploy wheres_alex_v009.aleo` gave no errors -- just a txn id and never seemed to land

But trying to deploy through aleo.tools spit back logs that it expected 3 outputs but only found 2. See pic attached for that error message
![image](https://github.com/AleoHQ/snarkOS/assets/39972641/50053a0d-e5dc-44c4-b794-4b45a4ed2f69)
